src/blindpsbt.cpp

```diff
@@ -311,8 +311,17 @@ bool ComputeAndAddToScalarOffset(uint256& a, CAmount value, const uint256& asset
     if (a.IsNull()) {
         a = scalar;
     } else {
-        // If we have a, then add the scalar to it.
-        if (secp256k1_ec_privkey_tweak_add(secp256k1_blind_context, a.begin(), scalar.begin()) != 1) return false;
+        uint256 scalar_negated = scalar;
+        if (secp256k1_ec_seckey_negate(secp256k1_blind_context, scalar_negated.begin()) != 1) {
+            return false;
+        }
+        // Special-case zero, which would otherwise cause `secp256k1_ec_privkey_tweak_add` to fail
```

Comment:
I don't really want to study this in the depth that would be required for me to check it, so: promise me you've thought really hard about why it's okay for this to be zero? (I assume it's not a privkey and can never be accidentally used as a privkey, despite the name of the function being used to do math on it.)

Reply:
Yes.